lib/GenomeFileUtil/core/GenbankToGenome.py

```diff
@@ -177,50 +163,391 @@ def stage_input(self, params):
+                "",  # web address
+                "",  # date
+                in_pub.authors,
+                in_pub.journal,
+            ]
+            date_match = re.match("\((\d{4})\)", in_pub.journal)
+            if date_match:
+                out_pub[4] = date_match.group(1)
+            if in_pub.pubmed_id:
+                out_pub[0:4] = [
+                    int(in_pub.pubmed_id),
+                    "PubMed",
+                    in_pub.title,
+                    "http://www.ncbi.nlm.nih.gov/pubmed/{}".format(
+                        in_pub.pubmed_id)]
+            pub_list.append(tuple(out_pub))
+        self.log("Parsed {} publication records".format(len(pub_list)))
+        return set(pub_list)
+
+    def _get_ontology(self, feature):
+        ontology = defaultdict(dict)
+        for key in ("GO_process", "GO_function", "GO_component"):
+            if key in feature.qualifiers:
+                sp = feature.qualifiers[key][0][3:].split(" - ")
+                ontology['GO'][sp[0]] = [1]
+                self.ontologies_present['GO'][sp[0]] = sp[1]
+        # TODO: Support other ontologies
+        return dict(ontology)
+
+    def _parse_features(self, record, source):
+        def _get_id(feat, tags=None):
+            _id = ""
+            if not tags:
+                tags = ['locus_tag', 'gene']
+            while tags and not _id:
+                _id = in_feature.qualifiers.get(tags.pop(0), [""])[0]
+            return _id
+
+        def _location(seq, feat):
+            strand_trans = ("", "+", "-")
+            loc = []
+            for part in feat.location.parts:
+                if part.strand >= 0:
+                    begin = int(part.start)
+                else:
+                    begin = int(part.end)
+                loc.append((
+                        record.id,
+                        begin + 1,
+                        strand_trans[part.strand],
+                        len(part)))
+            return loc
+
+        def _aliases(feat):
+            keys = ('locus_tag', 'old_locus_tag', 'protein_id')
+            alias_list = []
+            for key, val_list in feat.qualifiers.items():
+                if key in keys:
+                    alias_list.extend([(key, val) for val in val_list])
+            return alias_list
+
+        def _is_parent(feat1, feat2):
+            def _contains(loc1, loc2):
+                if loc1[2] != loc2[2]:  # different strands
+                    return False
+                elif loc1[2] == "+":
+                    return loc2[1] >= loc1[1] and (
+                    loc2[1] + loc2[3] <= loc1[1] + loc1[3])
+                else:
+                    return loc2[1] <= loc1[1] and (
+                    loc2[1] - loc2[3] >= loc1[1] - loc1[3])
+
+            # if any location in the CDS in contained by any location of the gene
+            return any([_contains(x, y) for x, y in
+                        itertools.product(feat1['location'], feat2['location'])])
+
```

Comment:
This logic only works for Genes as a parent (and only if NOT transpliced). 
If Transpliced we are kind of stuck going positionally in the file.
We need much more involved logic to check mRNA parentage. Information in the PD.

Reply:
What is the plan for CDS and mRNA relationships?